### PR TITLE
Integrate changes for Flutter Forward and Flutter 3.7

### PR DESCRIPTION
**Pull request by parlough**
Incorporates changes made for Flutter 3.7 and Flutter Forward.

Supersedes https://github.com/flutter/website/pull/8154 with an interactive rebase on `main` to fix the history issue, as well as an extra build fix.

### CODE REVIEW

**Comment by parlough** [comment]
PTAL @sfshaza2

File: .github/workflows/link-check.yml [hunk 1]
```diff
@@ -1,12 +1,14 @@
-name: Validate link references in built HTML
+name: Check links and link references
 
 on:
   push:
     branches:
       - main
+      - flutter-forward-23
```
Comment (domesticmouse) [inline .github/workflows/link-check.yml:7]: This can probably go away as I expect this branch name won't be reused

**Review by domesticmouse — APPROVED** [review]

File: .github/workflows/link-check.yml (hunk 1 above)
Comment (parlough) [reply .github/workflows/link-check.yml]: Nice catch. Removed :)

**Review by khanhnwin — APPROVED** [review]

**Comment by khanhnwin** [comment]
Thank. You. @parlough!!!

**Pull request merged**